cherrypy/_cprequest.py

```diff
@@ -929,6 +929,11 @@ def finalize(self):
             # and 304 (not modified) responses MUST NOT
             # include a message-body."
             dict.pop(headers, 'Content-Length', None)
+            # self.body is a generator object that must be yielded
+            # for _caching.tee_output() to finish it's job and
+            # remove orphaned threading._Event object from cache
+            for i in self.body:
```

Comment:
This could be replaced by `self.collapse_body()`, but I'm still trying to figure out a fix on a different level.